js-project-guidelines.md

```diff
@@ -11,30 +13,144 @@ The creation of this document came as a result of the hurdles that the IPFS comm
 
 Out goals for each JavaScript project are:
 
-- Be browser compatible, if possible (exceptions are for modules that require native bindings).
-- Not break CommonJS `require`. This means that if someone requires a JavaScript module from the IPFS ecosystem, they should be able to require and use browserify, webpack and other bundlers directly, without having to worry how much shims have to be added to make it work.
-- Make the UX of contributing and developing great with good guides, saving a lot of time in discussions that can be used for productivity.
+- Should be browser compatible, possible exceptions here are
+  - access to the file system
+  - native bindings.
```

Comment:
good point. Hard to put it in a such short sentence, what about giving examples:

`transport (http, tcp, udp, udt, utp, webrtc, websockets, etc)`

This should ring the bell, right?